src/main/java/uk/gov/companieshouse/pscdataapi/transform/CompanyPscTransformer.java

```diff
@@ -0,0 +1,110 @@
+            Updated updated = new Updated();
+            updated.setAt(requestBody.getInternalData().getUpdatedAt());
+            pscDocument.setCreated(created);
+            pscDocument.setUpdatedBy(requestBody.getInternalData().getUpdatedBy());
+            pscDocument.setUpdated(updated);
+            pscDocument.setSensitiveData(requestBody.getExternalData().getSensitiveData());
+            manageDataFields(requestBody, pscDocument);
+        } catch (Exception exception) {
+            throw new FailedToTransformException(String.format(
+                    "Failed to transform API payload: %s", exception.getMessage()));
+        }
+
+        return pscDocument;
+    }
+
+    private void manageDataFields(FullRecordCompanyPSCApi requestBody, PscDocument pscDocument) {
+        PscData data = new PscData();
+
+        data.setAddress(requestBody.getExternalData().getData().getServiceAddress());
+        data.setCeasedOn(requestBody.getExternalData().getData().getCeasedOn());
+        data.setDescription(requestBody.getExternalData().getData().getDescription());
+        data.setEtag(requestBody.getExternalData().getData().getEtag());
+        data.setKind(requestBody.getExternalData().getData().getKind());
+        data.setLinks(requestBody.getExternalData().getData().getLinks());
+        data.setName(requestBody.getExternalData().getData().getName());
+        data.setNationality(requestBody.getExternalData().getData().getNationality());
+        data.setNaturesOfControl(requestBody.getExternalData().getData().getNaturesOfControl());
+        data.setResidentialAddressIsSameAsServiceAddress(requestBody.getExternalData()
+                    .getData().getResidentialAddressIsSameAsServiceAddress());
+        data.setSanctioned(requestBody.getExternalData().getData().getIsSanctioned());
+        data.setServiceAddressIsSameAsRegisteredOfficeAddress(requestBody.getExternalData()
+                    .getData().getServiceAddressSameAsRegisteredOfficeAddress());
+
+        String kind = requestBody.getExternalData().getData().getKind();
+
+        if (IndividualPscRoles.includes(kind)) {
+            handleIndividualFields(requestBody, data);
+        }
+        if (SecurePscRoles.includes(kind)) {
+            handleSecureFields(requestBody, data);
+        }
+        pscDocument.setData(data);
+    }
+
+    private void handleIndividualFields(FullRecordCompanyPSCApi requestBody, PscData data) {
+        createNameElements(requestBody, data);
+        data.setCountryOfResidence(requestBody.getExternalData().getData().getCountryOfResidence());
+    }
+
+    private void handleSecureFields(FullRecordCompanyPSCApi requestBody, PscData data) {
+        Boolean ceased = requestBody.getExternalData().getData().getCeasedOn() != null;
+        data.setCeased(ceased);
+    }
+
+    private void createNameElements(FullRecordCompanyPSCApi requestBody, PscData data) {
```

Comment:
Again same principle, have this return the name elements object, and set in the above object.